code/controllers/subsystem/timer.dm

```diff
@@ -379,6 +379,16 @@ SUBSYSTEM_DEF(timer)
 	next.prev = src
 	prev.next = src
 
+/datum/controller/subsystem/timer/proc/TimerTimeLeft(id)
+	if(!id)
+		CRASH("Bad timer ID!")
+	
+	var/datum/timedevent/T = timer_id_dict[id]
+	if(!T)
+		return 0
+	
+	return world.time - T.timeToRun
```

Comment:
You should also be checking if its spent. Because the qdel loop is after like 5 check_ticks, timers can stick around up to one minute after being spent before being qdeleted during high load.

Oh and client time timers are a thing. =P